src/elm/interval.coffee

```diff
@@ -235,18 +235,18 @@ module.exports.Collapse = class Collapse extends Expression
         if typeof b.low?.durationBetween == 'function'
           # handle DateTimes using durationBetween
           if a.high?.after b.low # overlap
-            a.high = b.high if !b.high? or b.high.after a.high
+            a.high = b.high if (!b.high? and b.low.before a.high) or b.high.after a.high
           else if a.high?.durationBetween(b.low, perWidth.unit).high <= perWidth.value
             a.high = b.high
-          else if a.high? and !b.high?
+          else if (a.high? and !b.high?) and (b.low.before a.high)
```

Comment:
Is this `else if` block even necessary?  If `a` and `b` pass this `else if`, wouldn't they have logically already passed the initial `if` statement?  In other words, in what case would `a.high?.after b.low` (or `a.high?.sameOrAfter b.low`) be `false`, but `(a.high? and !b.high?) and (b.low.before a.high)` be `true`?  I can't think of a case, so I think this might be dead code (and safely deleted)?

Reply:
Doesn't look necessary, and it's not hit by the tests even because it is hitting  conditions in the lines above. Will delete dead code.